.gitignore

```diff
@@ -1,9 +1,32 @@
-.DS_Store
 bin
-.idea
 
 nri-kubernetes.yaml
 nri-kubernetes.yml
 
 charts/internal/e2e-resources/Chart.lock
 charts/internal/e2e-resources/charts/
+
+# Downloaded chart dependencies
+**/charts/*.tgz
+
+# OSX trash
+.DS_Store
+
+# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
+.idea/
+*.iml
+
+# Vscode files
+.vscode
+
+# Emacs save files
+*~
+\#*\#
+.\#*
+
+# Vim-related files
+[._]*.s[a-w][a-z]
+[._]s[a-w][a-z]
+*.un~
+Session.vim
+.netrwhist
```

Comment:
nit: something I learn a few month back, there is a way to locally ignore this for all projects. https://stackoverflow.com/questions/7335420/global-git-ignore